src/mbgl/style/expression/compound_expression.cpp

```diff
@@ -215,14 +217,12 @@ struct SignatureType<Lambda, std::enable_if_t<std::is_class<Lambda>::value>>
 { using Type = typename SignatureType<decltype(&Lambda::operator())>::Type; };
 
 template <typename Fn>
-static std::unique_ptr<detail::SignatureBase> makeSignature(Fn evaluateFunction, std::string name) {
+static std::unique_ptr<detail::SignatureBase> makeSignature(std::string name, Fn evaluateFunction) {
```

Comment:
in future , we might want to use `const char*` instead of `string` here for efficiency